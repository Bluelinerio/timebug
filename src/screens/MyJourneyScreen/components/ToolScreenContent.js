//@flow
import React      from 'react'
import { View }   from 'react-native'
import ToolButton from '../containers/Tools/ToolButtonContainer'
import styles     from '../styles'

type ToolScreenContentProps = {
  steps: Array<any>,
  stepColors: any,
  tools: any,
}

class ToolScreenContent extends React.PureComponent<ToolScreenContentProps> {
  render() {
    const { steps, tools, stepColors } = this.props
    return (
      <View style={[styles.container, styles.content]}>
        {tools &&
          tools.map(tool => {
            const step = steps[`${tool.number}`]
            return (
              <ToolButton
                key={tool.key}
                tool={tool}
                step={step}
                stepColors={stepColors}
              />
            )
          })}
      </View>
    )
  }
}

export default ToolScreenContent
