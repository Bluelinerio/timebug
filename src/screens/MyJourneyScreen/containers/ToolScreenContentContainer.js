//@flow
import invariant            from 'invariant'
import { connect }          from 'react-redux'
import { compose }          from 'recompose'
import selectors            from '2020_redux/selectors'
import { getUnlockedTools } from '2020_services/tools'
import ToolScreenContent    from '../components/ToolScreenContent'

// TODO: Work from here on, remove steps and any instance of these from every level
const mapStateToProps = (state: any) => {
  const steps = selectors.steps(state)
  const stepColors = selectors.statefullStepColors(state)

  if (__DEV__) {
    invariant(
      stepColors,
      `the colors for completed-uncompleted steps is not defined`
    )
  }

  const completedSteps = selectors.getCompletedSteps(state)

  const tools = getUnlockedTools(completedSteps)

  return {
    steps,
    stepColors,
    tools,
  }
}

export default compose(connect(mapStateToProps))(ToolScreenContent)
