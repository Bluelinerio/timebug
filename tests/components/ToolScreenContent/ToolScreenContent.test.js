import React from 'react'
import { shallow } from 'enzyme'
import { createMockStore } from 'redux-test-utils'
import shallowWithStore from '../../shallowWithStore'
import ToolScreenContentContainer from '../../../src/screens/MyJourneyScreen/containers/ToolScreenContentContainer'
import ToolScreenContent from '../../../src/screens/MyJourneyScreen/components/ToolScreenContent'

describe('Connected Tool screen content component', function() {
  it('Should render succesfully the connected component', function() {
    const dummyData = {
      cms: {
        steps: {
          1: {
            number: 1,
            icon: {
              uri:
                'https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg',
            },
          },
          2: {
            number: 2,
            icon: {
              uri:
                'https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg',
            },
          },
          3: {
            number: 3,
            icon: {
              uri:
                'https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg',
            },
          },
          4: {
            number: 4,
            icon: {
              uri:
                'https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg',
            },
          },
          5: {
            number: 5,
            icon: {
              uri:
                'https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg',
            },
          },
          6: {
            number: 6,
            icon: {
              uri:
                'https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg',
            },
          },
        },
      },
    }
    const store = createMockStore(dummyData)
    const component = shallowWithStore(<ToolScreenContentContainer />, store)
    expect(component).toMatchSnapshot()
  })

  it('Should render the component itself', function() {
    const steps = {
      1: 'step_1',
      2: 'step_2',
      3: 'step_3',
    }
    const tools = [
      {
        number: 1,
      },
      {
        number: 2,
      },
      {
        number: 3,
      },
    ]
    const stepColors = {}
    const component = shallow(
      <ToolScreenContent steps={steps} tools={tools} stepColors={stepColors} />
    )
    expect(component.dive().find('View').first().prop('children').length).toBe(3)
    expect(component).toMatchSnapshot()
  })
})
