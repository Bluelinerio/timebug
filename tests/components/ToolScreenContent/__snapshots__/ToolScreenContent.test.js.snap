// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Tool screen content component Should render succesfully the connected component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(ToolScreenContent) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "stepColors": Object {
        "PHASE1COMPLETE": "#4EADFF",
        "PHASE1INCOMPLETE": "#B8DEFF",
        "PHASE2COMPLETE": "#F89A1F",
        "PHASE2INCOMPLETE": "#FAB152",
        "PHASE3COMPLETE": "#20B89D",
        "PHASE3INCOMPLETE": "#20DE9D",
      },
      "steps": Object {
        "1": Object {
          "icon": Object {
            "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
          },
          "number": 1,
        },
        "2": Object {
          "icon": Object {
            "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
          },
          "number": 2,
        },
        "3": Object {
          "icon": Object {
            "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
          },
          "number": 3,
        },
        "4": Object {
          "icon": Object {
            "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
          },
          "number": 4,
        },
        "5": Object {
          "icon": Object {
            "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
          },
          "number": 5,
        },
        "6": Object {
          "icon": Object {
            "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
          },
          "number": 6,
        },
      },
      "tools": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "stepColors": Object {
          "PHASE1COMPLETE": "#4EADFF",
          "PHASE1INCOMPLETE": "#B8DEFF",
          "PHASE2COMPLETE": "#F89A1F",
          "PHASE2INCOMPLETE": "#FAB152",
          "PHASE3COMPLETE": "#20B89D",
          "PHASE3INCOMPLETE": "#20DE9D",
        },
        "steps": Object {
          "1": Object {
            "icon": Object {
              "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
            },
            "number": 1,
          },
          "2": Object {
            "icon": Object {
              "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
            },
            "number": 2,
          },
          "3": Object {
            "icon": Object {
              "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
            },
            "number": 3,
          },
          "4": Object {
            "icon": Object {
              "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
            },
            "number": 4,
          },
          "5": Object {
            "icon": Object {
              "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
            },
            "number": 5,
          },
          "6": Object {
            "icon": Object {
              "uri": "https://images.ctfassets.net/6h184bey8vl3/Rknt0PorAIG4uoMQAg4KM/d697d93618b298ce10d1606c7d3f573a/TimeBug-Icons-_Step-1_.svg",
            },
            "number": 6,
          },
        },
        "tools": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;

exports[`Connected Tool screen content component Should render the component itself 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToolScreenContent
    stepColors={Object {}}
    steps={
      Object {
        "1": "step_1",
        "2": "step_2",
        "3": "step_3",
      }
    }
    tools={
      Array [
        Object {
          "number": 1,
        },
        Object {
          "number": 2,
        },
        Object {
          "number": 3,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Connect(ToolButton)
          step="step_1"
          stepColors={Object {}}
          tool={
            Object {
              "number": 1,
            }
          }
        />,
        <Connect(ToolButton)
          step="step_2"
          stepColors={Object {}}
          tool={
            Object {
              "number": 2,
            }
          }
        />,
        <Connect(ToolButton)
          step="step_3"
          stepColors={Object {}}
          tool={
            Object {
              "number": 3,
            }
          }
        />,
      ],
      "style": Array [
        Object {
          "flex": 1,
        },
        Object {
          "padding": 20,
        },
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "step": "step_1",
          "stepColors": Object {},
          "tool": Object {
            "number": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "step": "step_2",
          "stepColors": Object {},
          "tool": Object {
            "number": 2,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "step": "step_3",
          "stepColors": Object {},
          "tool": Object {
            "number": 3,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Connect(ToolButton)
            step="step_1"
            stepColors={Object {}}
            tool={
              Object {
                "number": 1,
              }
            }
          />,
          <Connect(ToolButton)
            step="step_2"
            stepColors={Object {}}
            tool={
              Object {
                "number": 2,
              }
            }
          />,
          <Connect(ToolButton)
            step="step_3"
            stepColors={Object {}}
            tool={
              Object {
                "number": 3,
              }
            }
          />,
        ],
        "style": Array [
          Object {
            "flex": 1,
          },
          Object {
            "padding": 20,
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "step": "step_1",
            "stepColors": Object {},
            "tool": Object {
              "number": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "step": "step_2",
            "stepColors": Object {},
            "tool": Object {
              "number": 2,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "step": "step_3",
            "stepColors": Object {},
            "tool": Object {
              "number": 3,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
